content/sensu-go/5.17/reference/secret-providers.md

```diff
@@ -0,0 +1,297 @@
+-------------|------
+description  | Provider name used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "vault"{{< /highlight >}}
+
+### Spec attributes
+
+client       | 
+-------------|------ 
+description  | Map that includes secrets provider configuration [client attributes][12].
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"client": {
+  "address": "https://vaultserver.example.com:8200",
+  "token": "VAULT_TOKEN",
+  "version": "v1",
+  "tls": {
+    "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem"
+}
+{{< /highlight >}}
+
+max_retries  | 
+-------------|------ 
+description  | Number of times to retry connecting to the vault provider.
+required     | true
+type         | Integer
+default      | 2
+example      | {{< highlight shell >}}"max_retries": 2{{< /highlight >}}
+
+timeout      | 
+-------------|------ 
+description  | Provider connection timeout (hard stop).
+required     | false
+type         | String
+default      | 60s
+example      | {{< highlight shell >}}"timeout": "20s"{{< /highlight >}}
+
+rate_limiter | 
+-------------|------ 
+description  | Maximum rate and burst limits for the secrets API.
+required     | false
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"rate_limiter": {
+  "limit": 10.0,
+  "burst": 100
+}
+{{< /highlight >}}
+
+#### Client attributes
+
+address      | 
+-------------|------ 
+description  | Vault server address.
+required     | true
+type         | String
+example      | {{< highlight shell >}}
+"address": "https://vaultserver.example.com:8200"
+{{< /highlight >}}
+
+token        | 
+-------------|------ 
+description  | Vault token to use for authentication.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"token": "VAULT_TOKEN"{{< /highlight >}}
+
+version      | 
+-------------|------ 
+description  | HashiCorp Vault [HTTP API version][14].
+required     | true
+type         | String
+example      | {{< highlight shell >}}"version": "v1"{{< /highlight >}}
+
+tls          | 
+-------------|------ 
+description  | TLS object. Vault only works with TLS configured. You may need to set up a CA cert if it is not already stored in your operating system's trust store. To do this, set the TLS object, and provide the `ca_cert` path. You may also need to set up a client cert, client key, or cname.
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"tls": {
+  "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem",
+  "client_cert": "/etc/ssl/certs/vault_cert.pem",
+  "client_key": "/etc/ssl/certs/vault_key.pem",
+  "cname": "vault_client.example.com"
+}
+{{< /highlight >}}
+
+#### Rate limiter attributes
+
+limit        | 
+-------------|------ 
+description  | Maximum number of secrets requests per second that can be transmitted to the backend with the agent secrets API.
```

Comment:
```suggestion
description  | Maximum number of secrets requests per second that can be transmitted to the backend with the agent secrets API.
```

At the outset I assumed rate limiting would apply to communication with the Vault server, but "agent secrets API" makes it sound like we're configuring limits that apply to sensu-agents. Can we make this more clear?

Reply:
I believe it should just be "secrets API" (as in Sensu's secrets API).